beacon_node/network/src/sync/block_lookups/mod.rs

```diff
@@ -217,19 +220,29 @@ impl<T: BeaconChainTypes> BlockLookups<T> {
         // Make sure this block is not already downloaded, and that neither it or its parent is
         // being searched for.
         if let Some(parent_lookup) = self.parent_lookups.iter_mut().find(|parent_req| {
-            parent_req.contains_block(&parent_root) || parent_req.is_for_block(parent_root)
+            block_root == parent_req.chain_hash()
+                || parent_req.contains_block(&parent_root)
+                || parent_req.is_for_block(parent_root)
```

Comment:
In what scenario would line 223 and line 225 evaluate differently?

if the chain hash matches the `block_root`, the `current_parent_request` block root would always been `parent_root`, no?

Reply:
I was thinking similar to this: https://github.com/sigp/lighthouse/pull/5488/files#r1540337474

But this seems pretty clear that the `current_parent_request` and parent block downloads happen together in `add_child_components`, so I can revert this one.